files/build_templates/sonic_debian_extension.j2

```diff
@@ -0,0 +1,115 @@
+trap_push clean_sys
+sudo LANG=C chroot $FILESYSTEM_ROOT mount sysfs /sys -t sysfs
+
+sudo chroot $FILESYSTEM_ROOT service docker start
+sudo chroot $FILESYSTEM_ROOT docker version
+
+# Install config engine dependencies
+sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y install \
+    python-lxml    \
+    python-jinja2  \
+    python-netaddr \
+    python-ipaddr  \
+    python-yaml
+
+sudo dpkg --root=$FILESYSTEM_ROOT -i {{config_engine}}
+
+# Create all needed directories
+sudo mkdir -p $FILESYSTEM_ROOT/etc/ssw/
+sudo mkdir -p $FILESYSTEM_ROOT/etc/sonic/
+sudo mkdir -p $FILESYSTEM_ROOT/etc/sonic/templates/
+
+# Apply apt configuration files
+sudo cp $IMAGE_CONFIGS/apt/sources.list $FILESYSTEM_ROOT/etc/apt/
+sudo cp -R $IMAGE_CONFIGS/apt/sources.list.d/ $FILESYSTEM_ROOT/etc/apt/
+cat $IMAGE_CONFIGS/apt/sonic-dev.gpg.key | sudo LANG=C chroot $FILESYSTEM_ROOT apt-key add -
+
+# Apply environtment configuration files
+sudo cp $IMAGE_CONFIGS/environment/environment $FILESYSTEM_ROOT/etc/
+sudo cp $IMAGE_CONFIGS/environment/motd $FILESYSTEM_ROOT/etc/
+
+# Copy default minigraph
+sudo cp $IMAGE_CONFIGS/minigraph/minigraph_{{sonic_hwsku}}.xml $FILESYSTEM_ROOT/etc/sonic/minigraph.xml
+
+# Copy NTP configuration files and templates
+sudo cp $IMAGE_CONFIGS/ntp/ntp-config.service $FILESYSTEM_ROOT/etc/systemd/system/
+sudo LANG=C chroot $FILESYSTEM_ROOT systemctl enable ntp-config.service
+sudo cp $IMAGE_CONFIGS/ntp/ntp-config.sh $FILESYSTEM_ROOT/usr/bin/
+sudo cp $IMAGE_CONFIGS/ntp/ntp.conf.j2 $FILESYSTEM_ROOT/etc/sonic/templates/
+sudo cp $IMAGE_CONFIGS/ntp/ntp.yml $FILESYSTEM_ROOT/etc/sonic/
+
+# Copy rsyslog configuration files and templates
+sudo cp $IMAGE_CONFIGS/rsyslog/rsyslog-config.service  $FILESYSTEM_ROOT/etc/systemd/system/
+sudo LANG=C chroot $FILESYSTEM_ROOT systemctl enable rsyslog-config.service
+sudo cp $IMAGE_CONFIGS/rsyslog/rsyslog-config.sh $FILESYSTEM_ROOT/usr/bin/
+sudo cp $IMAGE_CONFIGS/rsyslog/rsyslog.conf.j2 $FILESYSTEM_ROOT/etc/sonic/templates/
+sudo cp $IMAGE_CONFIGS/rsyslog/rsyslog.yml $FILESYSTEM_ROOT/etc/sonic/
+sudo cp $IMAGE_CONFIGS/rsyslog/rsyslog.d/* $FILESYSTEM_ROOT/etc/rsyslog.d/
+
+# Copy interfaces configuration files and templates
+sudo cp $IMAGE_CONFIGS/interfaces/interfaces-config.service  $FILESYSTEM_ROOT/etc/systemd/system/
+sudo LANG=C chroot $FILESYSTEM_ROOT systemctl enable interfaces-config.service
+sudo cp $IMAGE_CONFIGS/interfaces/interfaces-config.sh $FILESYSTEM_ROOT/usr/bin/
+sudo cp $IMAGE_CONFIGS/interfaces/*.j2 $FILESYSTEM_ROOT/etc/sonic/templates/
+
+# Copy SNMP configuration files
+sudo cp $IMAGE_CONFIGS/snmp/snmp.yml $FILESYSTEM_ROOT/etc/sonic/
+
+# Generate system desription file
+export git_revision=$(git rev-parse --short HEAD)
+export sonic_hwsku={{sonic_hwsku}}
+export debian_version=`cat $FILESYSTEM_ROOT/etc/debian_version`
+export kernel_version={{kversion}}
+j2 $BUILD_TEMPLATES/sysDescription.j2 > sysDescription 
+sudo mv sysDescription $FILESYSTEM_ROOT/etc/ssw/
```

Comment:
taoyu, can you take a look at this to make sure we are generate the correct sonic version here.